### PR TITLE
Add variadic arguments support for fuzzer

### PR DESCRIPTION
**Pull request by tanjialiang**
Added support for variadic arguments. "greatest()" was called in the fuzzer and revealed a bug in itself. A separate PR will be appended to fix the bug.

### CODE REVIEW

File: velox/expression/tests/ExpressionFuzzer.cpp [hunk 1]
```diff
@@ -44,6 +44,12 @@ DEFINE_int32(
     100,
     "The number of elements on each generated vector.");
 
+DEFINE_int32(
+    max_num_varargs,
```
Comment (xiaoxmeng) [inline velox/expression/tests/ExpressionFuzzer.cpp:48]: Shall we use velox_fuzzer_xxx as gflag prefix?

Comment (tanjialiang) [reply velox/expression/tests/ExpressionFuzzer.cpp:48]: I think it should be fine. The fuzzer runs in another process, that is not connected to the main program.

File: velox/expression/tests/ExpressionFuzzer.cpp [hunk 2]
```diff
@@ -409,14 +415,15 @@ class ExpressionFuzzer {
   }
 
   void appendConjunctSignatures() {
```
Comment (kagamiori) [inline velox/expression/tests/ExpressionFuzzer.cpp:417]: As discussed offline, we would like to control the probability of generating conjunct in expressions. Appending conjunct signatures to `signatures_` wouldn't allow this control. Do you plan to make the change in a separate PR?

File: velox/expression/tests/ExpressionFuzzer.cpp [hunk 3]
```diff
@@ -466,11 +473,20 @@ class ExpressionFuzzer {
 
   std::vector<core::TypedExprPtr> generateArgs(const CallableSignature& input) {
     std::vector<core::TypedExprPtr> inputExpressions;
-    inputExpressions.reserve(input.args.size());
+    auto variableArity = input.variableArity;
+    auto numVarArgs = !variableArity
+        ? 0
+        : folly::Random::rand32(FLAGS_max_num_varargs + 1, rng_);
+    inputExpressions.reserve(input.args.size() + numVarArgs);
```
Comment (kagamiori) [inline velox/expression/tests/ExpressionFuzzer.cpp:480]: There will be at most `FLAGS_max_num_varargs + 1` arguments including the original argument in the function signature. Is this intended? If so, could you clarify in the comment for this flag?

**Review by kagamiori — COMMENTED** [review]
Hi Jialiang, Thank you for adding this support! I looks good to me except a few questions.

File: velox/expression/tests/ExpressionFuzzer.cpp (hunk 2 above)
Comment (tanjialiang) [reply velox/expression/tests/ExpressionFuzzer.cpp:417]: Thanks @kagamiori  for helping with reviweing. Yeah that should be belong to another PR.

File: velox/expression/tests/ExpressionFuzzer.cpp (hunk 3 above)
Comment (tanjialiang) [reply velox/expression/tests/ExpressionFuzzer.cpp:480]: I think this should be self-explanatory. num_varargs means number of variadic arguments. For example func(A a, B b, ...) the number of variadic arguments would be the number of "..." we have as they are the variadics. I can add some comments for clarification though.

There will be at most FLAGS_max_num_varargs arguments instead of FLAGS_max_num_varargs + 1. The "+1" is for rand gen to gen [0, FLAGS_max_num_varargs] inclusive.

Comment (kagamiori) [reply velox/expression/tests/ExpressionFuzzer.cpp:483]: In the code above we create conjuctSignature by `conjunctSignature.args = {BOOLEAN(), BOOLEAN()}`. So I thought we would do `inputExpressions.reserve(2 + numVarArgs)` here. Am I missing anything?

File: velox/expression/tests/ExpressionFuzzer.cpp [hunk 4]
```diff
@@ -44,6 +44,15 @@ DEFINE_int32(
     100,
     "The number of elements on each generated vector.");
 
+DEFINE_int32(
+    max_num_varargs,
+    5,
+    "The maximum number of variadic arguments fuzzer will generate for "
+    "functions that accept variadic arguments. Fuzzer will generate a random "
```
Comment (kagamiori) [inline velox/expression/tests/ExpressionFuzzer.cpp:51]: Maybe something like "The maximum number of variadic arguments fuzzer will generate for functions that accept variadic arguments. Fuzzer will generate up to max_num_varargs arguments for the variadic list in addition to the regular arguments."

**Review by kagamiori — APPROVED** [review]
Looks good to me except a nit. Thank you for the explanations!

File: velox/expression/tests/ExpressionFuzzer.cpp [hunk 5]
```diff
@@ -466,11 +476,20 @@ class ExpressionFuzzer {
...
+    auto variableArity = input.variableArity;
+    auto numVarArgs = !variableArity
+        ? 0
+        : folly::Random::rand32(FLAGS_max_num_varargs + 1, rng_);
+    inputExpressions.reserve(input.args.size() + numVarArgs);
 
     for (const auto& arg : input.args) {
       inputExpressions.emplace_back(generateArg(arg));
     }
+    if (variableArity) {
```
Comment (pedroerp) [inline velox/expression/tests/ExpressionFuzzer.cpp:488]: nit: for readability, I'd move the variableArity check and resizing the input expressions vector to this block, but not a big deal.

Comment (tanjialiang) [reply velox/expression/tests/ExpressionFuzzer.cpp]: I removed the condition but I did not change it to vector.resize() because that will make the rest of the varargs the same.
